### PR TITLE
fix(publish): respect the `spec.publishId` in build action config

### LINKED ISSUE #5094: 0.13: [Bug]: Publish does not adhere to specified `publishId`
### Current Behavior

When running `garden publish skopeo` it does the following:
```
ℹ build.skopeo         → Tagging image with skopeo:v-8b7165e21a and gardendev/skopeo:v-8b7165e21a
ℹ build.skopeo         → Publishing image gardendev/skopeo:v-8b7165e21a...
```

(It ignores the tag specified in `publishId`)

### Expected behavior

```
ℹ build.skopeo         → Tagging image with skopeo:v-8b7165e21a and gardendev/skopeo:1.41.0-3
ℹ build.skopeo         → Publishing image gardendev/skopeo:1.41.0-3...
```

### Reproducible example

in the garden core `images/skopeo/` directory
```
kind: Module
type: container
name: skopeo
description: Used by the kubernetes provider for interacting with container registries within a cluster
image: gardendev/skopeo:1.41.0-3
dockerfile: Dockerfile
extraFlags: [ "--platform", "linux/amd64" ]
```

or an equivalent build action like

```
kind: Build
type: container
name: skopeo
description: Used by the kubernetes provider for interacting with container registries within a cluster
spec:
  publishId: gardendev/skopeo:1.41.0-3
  dockerfile: Dockerfile
  extraFlags: [ "--platform", "linux/amd64" ]
```

### Workaround

Specify the --tag option, then it correctly only publish the specified `--tag`:

```
 % garden publish skopeo --tag foobar
[...]
ℹ build.skopeo         → Tagging image with skopeo:v-8b7165e21a and gardendev/skopeo:foobar
ℹ build.skopeo         → Publishing image gardendev/skopeo:foobar...
```

### Your environment

* **OS:** macOS
* **How I'm running Kubernetes:** This is just Docker, no kubernetes involved

`garden version` 0.13.13

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #5094
Partially addresses #5581

**Special notes for your reviewer**:

### CODE REVIEW

File: core/test/integ/src/plugins/container/container.ts [hunk 1]
```diff
@@ -62,15 +64,23 @@ describe("plugins.container", () => {
...
     sinon.replace(containerHelpers, "actionHasDockerfile", async () => true)
-    sinon.replace(containerHelpers, "dockerCli", async () => ({
-      all: "test log",
-      stdout: testVersionedId,
-      stderr: "",
-      code: 0,
-      proc: <any>null,
-    }))
+
+    dockerCli = dockerCli = sinon.stub(containerHelpers, "dockerCli")
```
Comment (mkhq) [inline core/test/integ/src/plugins/container/container.ts:74]: Double assignment `dockerCli = dockerCli =`

**Review by mkhq — COMMENTED** [review]
Looks good, just a small comment

File: core/test/integ/src/plugins/container/container.ts (hunk 1 above)
Comment (stefreak) [reply core/test/integ/src/plugins/container/container.ts:74]: Good catch!

**Review by stefreak — APPROVED** [review]
🥳

**Review by stefreak — APPROVED** [review]

**Pull request merged**